app/controllers/concerns/application_activities_concern.rb

```diff
@@ -0,0 +1,9 @@
+module ApplicationActivitiesConcern
+  extend ActiveSupport::Concern
+
+  def recent_activities(course, number)
+    return if number < 0 || !number.is_a?(Integer)
+    PublicActivity::Activity.where(owner: course).where.not('key like ?', '%notification').
```

Comment:
In my opinion, these three fields are just three names. Actually they are the same. Each of them will use polymorphic to store an object. Trackable is for the object you want to track. In our use cases it should be the input object like level, achievement...  As for owner and recipient, I think notification is for a user and generated by course or something else. So I use recipient filed to store user object. If we will have some expending, maybe we just need to change owner filed.